techtv2ovs/utils.py

```diff
@@ -184,12 +192,63 @@ def process_thumbs(self, ttv_video):
                 self.output.write("Error: thumbnail {} for ttv video {}".format(src['Key'], ttv_video.ttv_id))
                 ttv_video.status = ImportStatus.ERROR
                 ttv_video.thumbnail_status = ImportStatus.ERROR
-                ttv_video.errors += '{}\n\n'.format(str(exc))
+                ttv_video.errors += '{}:{}\n\n'.format(str(exc), traceback.format_exc())
 
         if ttv_video.thumbnail_status != ImportStatus.ERROR:
             ttv_video.thumbnail_status = ImportStatus.COMPLETE
         ttv_video.save()
 
+    def process_captions(self, ttv_video):
+        """
+        Copy over any caption files for a TechTVVideo from Dropbox to S3.
+        Create a VideoSubtitle object for each.
+
+        Args:
+            ttv_video (TechTVVideo): The TechTVVideo object to process
+        """
+        ttv_video.subtitle_status = ImportStatus.CREATED
+        captions = [caption[0] for caption in mysql_query(
+            'SELECT srt_file_file_name from closed_caption_files where video_id = %s', [ttv_video.ttv_id],
+            self.connection
+        )]
+        for caption in captions:
+            try:
+                basename, _ = os.path.splitext(caption)
+                dbx = dropbox.Dropbox(os.getenv('DROPBOX_TOKEN'))
+                dbx_folder = os.getenv('DROPBOX_FOLDER')
+                try:
+                    _, response = dbx.files_download(os.path.join('{}/{}'.format(dbx_folder, caption)))
```

Comment:
Can you change `os.path.join` from `os.path.join('{}/{}',dbx_folder, caption)` to `os.path.join(dbx_folder, caption)`? And similarly for the except clause below